### DISCUSSION TITLE
Gfx100s owner

### DISCUSSION THREAD

**Discussion by adrianuseless** [gkoh/furble#18] [General]
Hi

I am not a tech person but I do own a gfx100s and would love a wireless remote

I have no clue about coding etc but would be happy to pay for one to be made and I would test....

Adrian

**Comment by gkoh** [comment]
Hello @adrianuseless,
I'm happy to try and help get you up and running.

A few questions:
* a: Does the GFX 100s work the Fujifilm Camera Remote app?
   * b: Does it work in shutter-only/bluetooth mode?
* c: Do you use Windows or Mac?

If yes to (a) and (b) chances are good the GFX 100s is already supported.

Depending on (c), I'll try to come up with a guide for non-tech people.
As noted in #9, we can do a lot more to make this project easier to access.

**Comment by gkoh** [comment]
@adrianuseless I just enabled discussions and migrated your issue over.
Hopefully it's easier to ask stuff like this in future.

**Comment by adrianuseless** [comment]
Hi & thanks for replying and offering help

Yes the 100s works with the fuji own app trigger
I am not sure what you mean by shutter / bluetooth mode

I use a Mac

thanks

Adrian

**Reply by gkoh** [reply]
Cool.
In the app, there’s a mode called “Remote Release” which only controls the shutter.
furble uses this shooting mode to operate over Bluetooth.
If that works, then we can certainly progress.

**Reply by adrianuseless** [reply]
I have just checked and yes it works! 
I have used that before

**Reply by gkoh** [reply]
Excellent! Thanks for checking and confirming.
So, step one is to get the hardware.
If it’s OK, please purchase one of these:
https://shop.m5stack.com/products/m5stickc-plus-esp32-pico-mini-iot-development-kit

Should be $19.95 USD, no other bits needed. Not sure what delivery costs are like these days.

I will work on simplifying the programming of the M5StickC over the next week or so.
Hopefully I’m ready by the time it arrives.

**Reply by adrianuseless** [reply]
That's great thanks!

I have ordered two and they should be here in next week

I will let you know when they arrive

**Comment by adrianuseless** [comment]
a silly question but could it ever be made to be inside an item like this?
https://www.wexphotovideo.com/pentax-o-rc1-weatherproof-remote-control-for-wg-3-1539505/?cpgnid=18138250963&adext=&account=853-353-2386&campaign=&group=&mkwid=_dc&pcrid=&kword=&match=&plid=&pgrid=&ptaid=&si=&gclid=CjwKCAiA9qKbBhAzEiwAS4yeDYjazCkBKRdf5BdCva881aNt89o2SftBrP0HtDaZ_PmuIDwjnH4AcRoCHj0QAvD_BwE

**Reply by gkoh** [reply]
Not silly, this was my initial concept target 🙂
The answer is yes, given enough time and engineering resources.
Unfortunately I have neither and the M5StickC was a pretty good fallback.
The biggest issue is the ESP32 microcontroller eats power. I can only get about 1-2 hours continuous remote trigger until the battery dies.
With something like a Nordic NRF chip, I'd maybe 10x that battery life.

**Comment by adrianuseless** [comment]
Thanks, a shame we can't just modify an existing remote with the fuji signal

the units arrive Monday 

thanks

**Comment by gkoh** [comment]
@adrianuseless Are you able to try installing [esphome-flasher](https://github.com/esphome/esphome-flasher) for Mac?
The release is here:
https://github.com/esphome/esphome-flasher/releases/download/1.4.0/ESPHome-Flasher-1.4.0-macOS.tar

I was able to download furble release v1.0.0 and program my M5StickC from a Windows 10 system.
I don't have a Mac, so don't know how well it works there.

**Comment by adrianuseless** [comment]
Hi

I get this message


￼

**Comment by adrianuseless** [comment]
The remotes have now arrived!

**Reply by gkoh** [reply]
Wow, that was fast!
I assume you can charge them and they appear to work?
M5 usually ship them with the demo software, which cycles through various examples of the peripherals.

**Comment by adrianuseless** [comment]
I have managed to open the software on the Mac[](url)

<img width="1204" alt="Screenshot 2022-11-12 at 13 05 43" src="https://user-images.githubusercontent.com/117640925/201475397-cba56fe8-e741-4a6a-9d6d-5109054ad84a.png">

**Reply by gkoh** [reply]
This looks good.
When the M5StickC is plugged in, it should appear in the 'Serial port' drop down. On Windows it was 'COM<something>", not sure about Mac. I'd expect only one serial device will appear.
For the 'Firmware' you need to download the image from the *furble* releases:
https://github.com/gkoh/furble/releases/tag/v1.0.0

In particular, you want:
https://github.com/gkoh/furble/releases/download/v1.0.0/furble-m5stick-c-plus-v1.0.0+1.bin

Once both fields are set, hit 'Flash ESP' and you should get a bunch of progress text, most of it should be human-readable :)

There should be a report of programming success or some error when done.
Please let me know how you go.

**Reply by gkoh** [reply]
If you do manage to confirm successful programming, I hope the interface is intuitive enough.
You will need to set the camera to pairing mode, then scan for it.
If the camera asks to set the date/time, cancel, *furble* does not currently support sending time sync information.

**Comment by adrianuseless** [comment]
Hi

Thanks for the information,

Do I plug the device into my computer and if so with which port?

Thanks

Adrian

**Reply by gkoh** [reply]
Yes, please plug the M5StickC into your computer via cable.
The M5StickC uses a USB type C connector, I think Mac's have these ports?
If so, you just need a type C -> type C cable.
(From memory, my M5StickC came with a USB type A -> type C short cable)

If you can get the right cable, perhaps one way to confirm it's visible is to open 'esphome-flasher' with nothing connected, click the 'Serial port' drop down to see what entries exist.
Then connect the M5StickC, hit the 'refresh' arrows button and see if the drop down list updates to include a new device, if so, that will be the M5StickC.

**Comment by adrianuseless** [comment]
Hi

I downloaded the item from this link 

https://github.com/gkoh/furble/releases/download/v1.0.0/furble-m5stick-c-plus-v1.0.0+1.bin

but not sure from your other link which option to choose


thanks

Adrian 
<img width="1204" alt="Screenshot 2022-11-13 at 08 01 49" src="https://user-images.githubusercontent.com/117640925/201512219-854c5a26-ed49-4ee0-97d1-d0ca0e683b01.png">

**Reply by gkoh** [reply]
Oh, the first link was to the full release page, which includes files for the original M5StickC (now discontinued) and the replacement M5StickC Plus which you now have.
The second link was a direct link to `furble-m5stick-c-plus-v1.0.0+1.bin`, which you have already correctly obtained, that's all you need from this project.

**Comment by adrianuseless** [comment]
so just the one install?

**Reply by gkoh** [reply]
OK, just to be extra clear:
1. Install and run `esphome-flasher`
   1. You already did this
2. Download `furble-m5stick-c-plus-v1.0.0+1.bin`
   1. You already did this
3. Browse for the above file for the `Firmware` bit in `esphome-flasher`

**Comment by adrianuseless** [comment]
It works!!

even when I turn off them camera and then back on it connects with one push

will try out today

**Reply by gkoh** [reply]
Hooray! :grin: 
Yes! The pairing is saved on both camera and furble.
I'm actually impressed we got it working for you :slightly_smiling_face:

**Reply by gkoh** [reply]
Just note that if you re-program with `esphome-flasher` it will wipe all the settings, requiring a re-pairing exercise.
You shouldn't need to do this very often though.

**Comment by adrianuseless** [comment]
Im more impressed an idiot like me did it!

hahahahaha

**Comment by adrianuseless** [comment]
Now all I need to sort is some kind of way to weatherproof and attach it to a lanyard etc

**Comment by adrianuseless** [comment]
Ive been testing it today and its good, it works upto about 10-15m and battery life is ok
charges back up quickly 

a few things I've noticed 

Is there a way to have focus set to trigger button too? and maybe an option for either?
a timer option for auto off? 2/5/10 min?

thanks

**Reply by gkoh** [reply]
Thanks for testing!

> Ive been testing it today and its good, it works upto about 10-15m and battery life is ok charges back up quickly

It only just occurred to me that we don't set the bluetooth transmit power, it is currently the default.
Looking at the doco, we could bump the power up to max, which should double the range.
I'm not sure what happens to battery life though, should make it configurable.

> a few things I've noticed
> 
> Is there a way to have focus set to trigger button too? and maybe an option for either?

Is your aim to require focus before triggering?
We could make this configurable, but perhaps it would be better to put your GFX into focus priority mode?

> a timer option for auto off? 2/5/10 min?

I'll look into supporting this as a configurable setting, thanks.

**Comment by adrianuseless** [comment]
Hi again

m5 stack said to maybe buy this instead?

https://shop.m5stack.com/collections/m5-controllers/products/stick-c

would that work or just better buying the other?

**Reply by gkoh** [reply]
Ha, the M5StickC was discontinued in favour of the M5StickC Plus.
Ah, this tweet explains a little:
https://twitter.com/M5Stack/status/1592110906028470273

"limited quantity", maybe they found a box in the warehouse?

The Plus (compared to the no-plus) has:
* same size
* larger screen
* larger battery
* larger price

I would not consider the M5StickC worth it given you already have M5StickC Plus.

**Comment by adrianuseless** [comment]
HiBluetooth I was not being critical I think that it’s great But could you add it as an option in the menu for people who want it ? Then it’s like a boost mode and depletes battery when used in boost mode?FocusI was asked if the take image button could also get focus too?If not then it’s not the end of the world tbhWhat Fuji cameras have you tested it on?I was going to go to my local store and test on some Adrian Sent from my iPhoneOn 15 Nov 2022, at 12:34 am, Guo-Rong ***@***.***> wrote:﻿
Ha, the M5StickC was discontinued in favour of the M5StickC Plus.
Ah, this tweet explains a little:
https://twitter.com/M5Stack/status/1592110906028470273
"limited quantity", maybe they found a box in the warehouse?
The Plus (compared to the no-plus) has:

same size
larger screen
larger battery
larger price

I would not consider the M5StickC worth it given you already have M5StickC Plus.

**Reply by gkoh** [reply]
> HiBluetooth I was not being critical I think that it’s great But could you add it as an option in the menu for people who want it ? Then it’s like a boost mode and depletes battery when used in boost mode?

I didn't take it as criticism, you merely prompted more thought.
I agree, it would useful to add an option to increase the transmit power.

> Focus I was asked if the take image button could also get focus too?If not then it’s not the end of the world tbh
I set focus priority on my Fuji so taking the image requires focus.
To do this with just the buttons is tricky, we can't fake the partial press tactility of the actual shutter.
The focus mode of button B requires you to hold it until focus is achieved, same as doing a partial shutter press, we can't do that with the single button A on the M5StickC.
We are effectively faking the 2 levels of shutter press on the actual camera with 2 separate buttons.

> What Fuji cameras have you tested it on?I was going to go to my local store and test on some

I have personally verified it works on my X-T30 and a friend's X100V.
Others in the community have verified with various X series (listed on the project page).
You are the first GFX owner to confirm it working :slightly_smiling_face:

**Comment by gkoh** [comment]
> What Fuji cameras have you tested it on?I was going to go to my local store and test on some

From what I've seen, I'm willing to bet it will work on all of them :grin:

**Comment by adrianuseless** [comment]
What are your plans with it?Adrian Sent from my iPhoneOn 15 Nov 2022, at 8:28 am, Guo-Rong ***@***.***> wrote:﻿

What Fuji cameras have you tested it on?I was going to go to my local store and test on some

From what I've seen, I'm willing to bet it will work on all of them 😁

**Reply by gkoh** [reply]
> What are your plans with it?

To be honest, none.
It was a fun hobby project that I thought might be interesting for others.

If I had time and money I would probably:
* build custom hardware
   * nice switch to emulate an actual shutter
   * weather proof enclosure
   * much lower power, eg. days of use
* support all the remaining major brands (eg. Nikon, Sony, Olympus, etc)

**Reply by gkoh** [reply]
Actually, I lie, there's one feature I would like to find some time to implement, intervalometer mode, particularly to support:
* n shots with m second delay
* long exposure mode, ie single shot with x millisecond open shutter

**Comment by adrianuseless** [comment]
I’ve been asked for some by owners tbh but I’m not a tech guyMy worry with investing is that Fuji could just bring one out tomorrow and that’s the end for it.. I’ve got a company giving me a quote for a silicone case to waterproof the item and add a lanyard attachment point and cover the external portsIf I get it sorted I’ll let you know how many I sell and of course cut you in.Does that sound fair?AdrianSent from my iPhoneOn 15 Nov 2022, at 9:08 am, Guo-Rong ***@***.***> wrote:﻿

What are your plans with it?
To be honest, none.
It was a fun hobby project that I thought might be interesting for others.

If I had time and money I would probably:

build custom hardware

nice switch to emulate an actual shutter
weather proof enclosure
much lower power, eg. days of use


support all the remaining major brands (eg. Nikon, Sony, Olympus, etc)

**Reply by gkoh** [reply]
> I’ve been asked for some by owners tbh but I’m not a tech guy

Ha, interesting, nice.

> My worry with investing is that Fuji could just bring one out tomorrow and that’s the end for it..

This is certainly a possibility, but Fuji would still need to support all those 'legacy' cameras in circulation today.
Their Bluetooth protocol is reasonably good, I'd be surprised if they dropped it suddenly.

> I’ve got a company giving me a quote for a silicone case to waterproof the item and add a lanyard attachment point and cover the external ports
> If I get it sorted I’ll let you know how many I sell and of course cut you in.Does that sound fair?

I've been asked before to enable the sponsorship system in GitHub, it would be totally acceptable to me to do that and you could donate whatever you feel is appropriate.
Thank you for the consideration.

**Comment by adrianuseless** [comment]
Do you know the guys from m5 stack? I was wondering if they would make a version for us that’s got less parts ?Adrian Sent from my iPhoneOn 15 Nov 2022, at 9:15 am, Adrian Snelius ***@***.***> wrote:﻿I’ve been asked for some by owners tbh but I’m not a tech guyMy worry with investing is that Fuji could just bring one out tomorrow and that’s the end for it.. I’ve got a company giving me a quote for a silicone case to waterproof the item and add a lanyard attachment point and cover the external portsIf I get it sorted I’ll let you know how many I sell and of course cut you in.Does that sound fair?AdrianSent from my iPhoneOn 15 Nov 2022, at 9:08 am, Guo-Rong ***@***.***> wrote:﻿

What are your plans with it?
To be honest, none.
It was a fun hobby project that I thought might be interesting for others.

If I had time and money I would probably:

build custom hardware

nice switch to emulate an actual shutter
weather proof enclosure
much lower power, eg. days of use


support all the remaining major brands (eg. Nikon, Sony, Olympus, etc)

**Reply by gkoh** [reply]
> Do you know the guys from m5 stack? I was wondering if they would make a version for us that’s got less parts ?

I do not know them, however they do seem to offer a design service, so it would be very possible to enquire what kind of costs may be involved.
It would be interesting to explore whether this path is feasible, economically.

**Comment by adrianuseless** [comment]
I’ve emailed them a few times but no reply which is frustrating Adrian Sent from my iPhoneOn 15 Nov 2022, at 10:48 am, Guo-Rong ***@***.***> wrote:﻿

Do you know the guys from m5 stack? I was wondering if they would make a version for us that’s got less parts ?

I do not know them, however they do seem to offer a design service, so it would be very possible to enquire what kind of costs may be involved.
It would be interesting to explore whether this path is feasible, economically.

**Comment by adrianuseless** [comment]
they replied and said yes to maybe a custom order

does this sound right for our needs?

email from me to them

Hi

I want the M5StickC Plus

For a camera remote

I do not need

ESP32-based support Wi-Fi
Built-in 6-Axis IMU
Red LED
IR transmitter
Microphone
RTC
Extendable Socket
Built-in Passive Buzzer
Wearable & Wall mounted

I am not sure what the etc is to be honest

I would like the main button to be used like a camera half push to focus and full to take if that’s possible?

I need the Usbc socket but no others

I would like it better sealed and with a lanyard point on the bottom where the grove socket would be

As you can see I don’t need all the capabilities I need it for the bluetooth mainly 

Is that possible?

Adrian

**Reply by gkoh** [reply]
Hi Adrian,

The 'maybe custom order' will likely be contingent on order quantity and price.
If you're looking to build 1000 units, they will likely be very helpful.

To preserve existing development, we need to keep the ESP32 microcontroller, this has bluetooth and wifi built into the silicon.
Changing microcontrollers would be significant development, for which I have not the resources.
(I would estimate hundred hours of development so far invested).

We would want:
* ESP32
   * external oscillator to support lower power bluetooth operation during sleep
* same form factor
* same OLED screen
* possibly larger capacity battery
* button A changed to dual action switch
* IPxx ingress protection

If you're serious about pursuing this, we should probably move this to a private email thread.

**Comment by adrianuseless** [comment]
my email is Adriansnelius@mac.com

**Comment by adrianuseless** [comment]
hi, have you tried to email?

**Reply by gkoh** [reply]
Apologies, life got in the way :slightly_smiling_face: 
Email sent, let's take it up there.
